### ISSUE TITLE
FreeBSD 13 based experimental image unbootable after installing

### ISSUE THREAD

**Issue by mbnunes** [helloSystem/hello#242]
gentlemen, good night.  

first, congratulations on the project!  i'm putting the experimental hellosystem 13 on my production computer, i need the video drive from AMD.  but I had to boot in verbose mode, it was stuck on the Splash screen, but my problem here and after installing hellosystem on my HD, after restarting, it shows the following screen:

![IMG_20211106_014914](https://user-images.githubusercontent.com/931636/140598673-d3af613c-6119-4ad5-94a5-4a0bba7a2fa3.jpg)

I tried looking for the problem here in the issues, I couldn't find it.  what can I do?

**Comment by probonopd** [comment]
This is unchartered territory. The FreeBSD 13-based experimental build is known to have this issue.

> what can I do?

Try to find out why it wokrs with the FreeBSD 12.2 based build but not the FreeBSD 13 based build.
I don't have more knowledge than anyone on these things, really. So experimentation, time, and debugging is in order :)

**Comment by woomia** [comment]
You got further than me, mine stops because it can't find a boot loader.

**Comment by probonopd** [comment]
Same for me on a non-EFI machine.

**Comment by woomia** [comment]
![7B927A0B-6C13-4544-8E39-0546A900F520](https://user-images.githubusercontent.com/1365979/140612236-4fc17fcc-57cb-40b2-90a0-b02b8b3cfc78.jpeg)

This is what happens when I try to booting from the ISO.

**Comment by mbnunes** [comment]
Thanks for your attention. I managed to install experimental 13 using EFI. No version 0.6.0 of amdgpu_pro driver related error. I will reinstall version 0.6.0 and already post here.

**Comment by probonopd** [comment]
I am getting the same issue in Legacy mode. Any insights into why this is happening would be appreciated.

**Comment by probonopd** [comment]
I note that it says "CD Loader" even though the system has been installed to the hard disk. Is that part of what is going wrong here? (Does a FreeBSD 12.2 helloSystem installed to disk also say "CD Loader" there?)

**Comment by woomia** [comment]
That error message is from the live ISO, it wasn't installed to the hard drive.

**Comment by probonopd** [comment]
Oops, right, I mixed it up.

**Comment by mbnunes** [comment]
@probonopd 
just to let you know, I did a test using the FreeBSD 13 release and the exact same thing happened, but you can see in the log that it tries to mount the root on usb, that is, this is not a project-only bug.

![1636417556140](https://user-images.githubusercontent.com/931636/140840506-a658ce22-e2ef-460b-b41c-7b48a73e2c2b.jpg)

**Comment by probonopd** [comment]
@mbnunes just to be 100% sure, when you say "using the FreeBSD 13 release" you mean the one from freebsd.org, __not__ helloSystem?

In any case, can you enter `?` there, press enter, and post the result?

Thanks.

**Comment by mbnunes** [comment]
Is possible install with UFS?

**Comment by probonopd** [comment]
Yes, but not with the graphical installer. Would need to be done by the user by hand.
Why would one want that, though?

**Comment by probonopd** [comment]
Thanks @jsm222 for finding the following out:

> Executing the following on hello-0.7.0_0G41-FreeBSD-13.0-amd64.iso after the installation,
> makes a bootable helloSystem efi system this time tested on qemu/kvm efi.

```
mount -t msdosfs /dev/ada0p1 /mnt/boot/efi
bsdinstall -D /tmp/somelogfile bootconfig
```

**Comment by mbnunes** [comment]
> Thanks @jsm222 for finding the following out:
> 
> > Executing the following on hello-0.7.0_0G41-FreeBSD-13.0-amd64.iso after the installation,
> > makes a bootable helloSystem efi system this time tested on qemu/kvm efi.
> 
> ```
> mount -t msdosfs /dev/ada0p1 /mnt/boot/efi
> bsdinstall -D /tmp/somelogfile bootconfig
> ```

Same erro... After set commands, reboot the system and go to mountroot's console

**Comment by probonopd** [comment]
Now trying with

```
# Fix unbootable FreeBSD 13
# https://github.com/helloSystem/hello/issues/242#issuecomment-965617810
if [ -n "${INSTALLER_DEVICE}" ] ; then
  mount -t msdosfs /dev/"${INSTALLER_DEVICE}p1" "${FSMNT}"/boot/efi
  bsdinstall -D /tmp/bootconfig.log bootconfig
  umount "${FSMNT}"/boot/efi
fi
```

in the installer. Fingers crossed...

**Comment by probonopd** [comment]
...works!

**Comment by mbnunes** [comment]
@probonopd Good afternoon, how are you? hope so!

I used the last ISO from Experimental 13 0G67 everything worked beautifully, the boot didn't need to activate verbose mode, the resolution was correctly identified at boot time. However after the installation (defaults to ZFS), the resolution is not correctly identified, and the mountroot error continues.
![1636987025440](https://user-images.githubusercontent.com/931636/141808720-cee45327-1649-4965-bcdc-1e97db0bb0fc.jpg)

**Comment by probonopd** [comment]
You are not installing to a USB device, are you? Please describe more about your boot device hardware setup.

**Comment by mbnunes** [comment]
no, i am installing on my ADA branded 250gb SSD.  booting via USB 8gb UEFI.  I can film the entire installation process if I want.

**Comment by probonopd** [comment]
Not needed yet, I will do more testing soon. Thanks!
